tools/transform.cpp

```diff
@@ -379,6 +383,46 @@ static void calculateAndInitConstants(Transform &t) {
       return false;
     return true;
   };
+  // Returns access size. 0 if no access, -1 if unknown
+  const uint64_t UNKNOWN = -1, NO_ACCESS = 0;
+  auto get_access_size = [&](const Instr &inst) -> uint64_t {
+    if (dynamic_cast<const Alloc *>(&inst) ||
+        dynamic_cast<const Malloc *>(&inst))
+      // They are uninitialized (no actual bytes written), so assume that they
+      // are not accessed.
+      // Calloc is different: it contains memset to zero
+      return NO_ACCESS;
+
+    Type *value_ty = nullptr;
+    unsigned align = 0;
+    if (auto st = dynamic_cast<const Store *>(&inst)) {
+      value_ty = &st->getValue()->getType();
+      align = st->getAlign();
+    } else if (auto ld = dynamic_cast<const Load *>(&inst)) {
+      value_ty = &ld->getType();
+      align = ld->getAlign();
+    }
+
+    if (value_ty) {
+      if (auto ity = dynamic_cast<const IntType *>(value_ty))
```

Comment:
value_ty->bits() should work for all types, incl pointers

Reply:
Pointer's bits() returns bits_bid + bits_offset (which is 80), so does not work here.